RIOT-OS/RIOT
# boards/p-nucleo-wb55: add initial support

### Contribution description

The board comes with a stm32wb55rg processor as well as a cortexm0+ co-processor dedicated to radio operation. It comes with a 802.15.4 and BLE radio. Support for these radios and the co-processor is still pending.

This PR adds initial support for p-nucleo-wb55 board. This board need the following patch to be flashed with openocd http://openocd.zylin.com/#/c/4933/.

This PR adds support for stm32wb CPU. This CPU is much like stm32l4 so much of the code base can be re-used.

Basic peripheral support is provided: i2c, lpuar, uart, rtt, rtc, spi, timer.

No support for the BLE and 802.15.4 radio are provided for now.

### Testing procedure

Run:

`python3 dist/tools/compile_and_test_for_board/compile_and_test_for_board.py . p-nucleo-wb55`

Check the board documentation.

**Edit**: since this board is merging with the stm32l4 clock code just to be safe run the above command on any stm32l4. Board

### Issues/PRs references


Depends on ~~#11897~~ and ~~#11898~~. The dependency with ~~#11896~~ could be removed.
